lib/ansible/executor/task_executor.py

```diff
@@ -134,7 +134,10 @@ def _clean_res(res):
                 if isinstance(res, UnsafeProxy):
                     return res._obj
                 elif isinstance(res, binary_type):
-                    return to_text(res, errors='surrogate_or_strict')
+                    try:
+                        return to_text(res, errors='surrogate_or_strict')
+                    except:
+                        return bytes(res)
```

Comment:
In this case, we can add a simple test on the diff size. If over a given size (maybe 1M?), we don't send the diff back showing only a message.